File: PI2025_DESCTOP/src/main/java/models/Recompense.java
package models;

public class Recompense {
    int id;
    String description;
    int cout_en_points;
    String disponibilite;

    public Recompense() {};
    public Recompense(int id, String description, int cout_en_points, String disponibilite) {
        this.id = id;
        this.description = description;
        this.cout_en_points = cout_en_points;
        this.disponibilite = disponibilite;
    }
    public Recompense(String description, int cout_en_points, String disponibilite) {
        this.description = description;
        this.cout_en_points = cout_en_points;
        this.disponibilite = disponibilite;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getCout_en_points() {
        return cout_en_points;
    }

    public void setCout_en_points(int cout_en_points) {
        this.cout_en_points = cout_en_points;
    }

    public String getDisponibilite() {
        return disponibilite;
    }

    public void setDisponibilite(String disponibilite) {
        this.disponibilite = disponibilite;
    }

    @Override
    public String toString() {
        return "Recompense{" +
                "id=" + id +
                ", description='" + description + '\'' +
                ", cout_en_points=" + cout_en_points +
                ", disponibilite='" + disponibilite + '\'' +
                '}';
    }
}

File: ProjetWebTrekSwap/src/Entity/Recompense.php
<?php

namespace App\Entity;

use App\Repository\RecompenseRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: RecompenseRepository::class)]
class Recompense
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    private ?string $description = null;

    #[ORM\Column]
    private ?int $cout_en_points = null;

    #[ORM\Column(length: 255)]
    private ?string $disponibilite = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): static
    {
        $this->description = $description;

        return $this;
    }

    public function getCoutEnPoints(): ?int
    {
        return $this->cout_en_points;
    }

    public function setCoutEnPoints(int $cout_en_points): static
    {
        $this->cout_en_points = $cout_en_points;

        return $this;
    }

    public function getDisponibilite(): ?string
    {
        return $this->disponibilite;
    }

    public function setDisponibilite(string $disponibilite): static
    {
        $this->disponibilite = $disponibilite;

        return $this;
    }
}
